packaging/opensuse/snapd.spec

```diff
@@ -168,83 +200,67 @@ export CFLAGS
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snapctl
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap-seccomp
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=default -ldflags '-extldflags "-static"' %{import_path}/cmd/snap-update-ns
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=default -ldflags '-extldflags "-static"' %{import_path}/cmd/snap-exec
 %if 0%{?with_test_keys}
-# The gobuild macro doesn't allow us to pass any additional parameters
-# so we we have to invoke `go install` here manually.
-export GOPATH=%{_builddir}/go:%{_libdir}/go/contrib
-export GOBIN=%{_builddir}/go/bin
-# Options used are the same as the gobuild macro does but as it
-# doesn't allow us to amend new flags we have to repeat them here:
-# -s: tell long running tests to shorten their build time
-# -v: be verbose
-# -p 4: allow parallel execution of tests
-# -x: print commands
-go install -s -v -p 4 -x -tags withtestkeys github.com/snapcore/snapd/cmd/snapd
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie -tags withtestkeys %{import_path}/cmd/snapd
 %else
-%gobuild cmd/snapd
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snapd
 %endif
 
-%gobuild cmd/snap
-%gobuild cmd/snapctl
-# build snap-exec and snap-update-ns completely static for base snaps
-# NOTE: openSUSE's golang rpm helpers pass -buildmode=pie, but glibc is not
-# built with -fPIC so it'll blow up during linking, need to pass
-# -buildmode=default to override this
-%gobuild -buildmode=default -ldflags=-extldflags=-static cmd/snap-exec
-%gobuild -buildmode=default -ldflags=-extldflags=-static cmd/snap-update-ns
-# build snap-seccomp
-%gobuild cmd/snap-seccomp
-
 # Build C executables
-%make_build -C cmd
+%make_build -C %{indigo_srcdir}/cmd
 
 %check
-%{gotest} %{import_path}/...
-%make_build -C cmd check
+# Run tests with fixed locale that is expected by unicode/color code.
+LC_ALL=C.UTF-8 GOPATH=%{indigo_gopath}:$GOPATH go test %{import_path}/...
+%make_build -C %{indigo_srcdir}/cmd check
 
 %install
-# Install all the go stuff
-%goinstall
-# TODO: instead of removing it move this to a dedicated golang package
-rm -rf %{buildroot}%{_libdir}/go
-# Move snapd, snap-exec, snap-seccomp and snap-update-ns into {_libexecdir}/snapd
-install -m 755 -d %{buildroot}%{_libexecdir}/snapd
-mv %{buildroot}%{_bindir}/snapd %{buildroot}%{_libexecdir}/snapd/snapd
-mv %{buildroot}%{_bindir}/snap-exec %{buildroot}%{_libexecdir}/snapd/snap-exec
-mv %{buildroot}%{_bindir}/snap-update-ns %{buildroot}%{_libexecdir}/snapd/snap-update-ns
-mv %{buildroot}%{_bindir}/snap-seccomp %{buildroot}%{_libexecdir}/snapd/snap-seccomp
+# Install all systemd and dbus units, and env files.
+%make_install -C %{indigo_srcdir}/data \
+		BINDIR=%{_bindir} \
+		LIBEXECDIR=%{_libexecdir} \
+		SYSTEMDSYSTEMUNITDIR=%{_unitdir} \
+		SNAP_MOUNT_DIR=%{snap_mount_dir}
```

Comment:
I think you could also set `SNAPD_ENVIRONMENT_FILE=%{_sysconfdir}/default/snapd`, provided openSUSE uses `/etc/default`, otherwise it ends up using `/etc/environment` which is a bit annoying.

Reply:
+1 but I'd like to make that as a separate improvement. This PR is what shipped last night.